### PR TITLE
chore(deps): bump go to v1.22.3

### PR DESCRIPTION
**Pull request by rach-id**
## Description

bump go to v1.22.3

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use
  [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: .github/workflows/check-generated.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
     steps:
       - uses: actions/setup-go@v4
         with:
-          go-version: "1.22.2"
+          go-version: "1.22.3"
```
Comment (rootulp) [inline .github/workflows/check-generated.yml:46]: [optional] we can save ourselves from doing this in the future if we instead use https://github.com/celestiaorg/celestia-app/blob/ce1453f6e706ad25d09e2346070bdf79718f867d/.github/workflows/test.yml#L11-L13

**Review by rootulp — APPROVED** [review]

File: .github/workflows/check-generated.yml (hunk 1 above)
Comment (rach-id) [reply .github/workflows/check-generated.yml:46]: sure, let's do it in a separate PR and backport it to v0.34.x-celestia

Comment (rootulp) [reply .github/workflows/check-generated.yml:46]: SGTM

Comment (staheri14) [inline .github/workflows/check-generated.yml:46]: Is it possible to define a global variable for the Go version and reference it throughout the code? This way, we won't have to update the Go version in multiple locations. Though, for a follow-up PR, not the current one.

**Review by staheri14 — APPROVED** [review]

File: .github/workflows/check-generated.yml (hunk 1 above)
Comment (rach-id) [reply .github/workflows/check-generated.yml:46]: https://github.com/celestiaorg/celestia-core/issues/1349

**Pull request merged**